ex5/ex5.c

```diff
@@ -15,8 +16,40 @@ char* msg2 = "hello world #2";
 char* msg3 = "hello world #3";
 
 int main(void)
+
 {
-    // Your code here
+    char inbuf[MSGSIZE];    // a buffer that will hold the incoming data that is being written
+    int p[2];               // a two-element array to hold the read and write file descriptors that are used by the pipe   
+    pid_t ChildP;
+  
+
+    // establish our pipe, passing it the p array so that it gets populated by the read and write file descriptors
+    if (pipe(p) < 0) {
+        fprintf(stderr, "pipe failed\n");
+        exit(1);
+    }
     
+     ChildP = fork();
+     if(childpid == 0)
```

Comment:
Did you mean ChildP here? The compiler does not recognize childpid, and you never use the ChildP variable so I think if you replace this it should be good.